colossalai/shardformer/modeling/bert.py

```diff
@@ -29,13 +29,25 @@
 from transformers.utils import logging
 
 from colossalai.pipeline.stage_manager import PipelineStageManager
+from colossalai.shardformer import ShardConfig
+from colossalai.shardformer.layer._operation import gather_forward_split_backward, split_forward_gather_backward
 
 
 class BertPipelineForwards:
     '''
     This class serves as a micro library for forward function substitution of Bert models
     under pipeline setting.
     '''
+    tp_shard_config: ShardConfig = None
+
+    @staticmethod
+    def set_shard_config(cls, shard_config):
```

Comment:
You'd better inject this when replace forward function. This Class is used like namespace.

Reply:
> You'd better inject this when replace forward function. This Class is used like namespace.

updated